### PR TITLE
feat: cloak telescope preview

### LINKED ISSUE #8: Can we have cloak active in previews from other plugins such as telescope
The plugin works great but doesn't seem to apply itself to previews such as when showing a project search in e.g. telescope the preview should also be cloaked if applicable.

### PR DESCRIPTION
**Pull request by eslam-allam**
closes #8 

- Using glob2regex to add consistency with bufenter autocmd.
- Added a cloak_telescope option for users who don't have preview enabled.
- updated docs.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -31,6 +31,8 @@ require('cloak').setup({
   cloak_length = nil, -- Provide a number if you want to hide the true length of the value.
   -- Whether it should try every pattern to find the best fit or stop after the first.
   try_all_patterns = true,
+  cloak_telescope = false,
```
Comment (laytan) [inline README.md:34]:
```suggestion
  -- Set to true to cloak Telescope preview buffers.
  cloak_telescope = true,
```

File: README.md [hunk 2]
```diff
@@ -31,6 +31,8 @@ require('cloak').setup({
...
+  -- Set to true to cloak Telescope preview buffers.
```
Comment (laytan) [inline README.md:35]:
```suggestion

```

File: README.md (hunk 1 above)
Comment (laytan) [reply README.md:34]: It makes a lot of sense to enable by default imo

File: lua/cloak/init.lua [hunk 3]
```diff
@@ -16,6 +16,7 @@ M.opts = {
   highlight_group = 'Comment',
   try_all_patterns = true,
   patterns = { { file_pattern = '.env*', cloak_pattern = '=.+' } },
+  cloak_telescope = false,
```
Comment (laytan) [inline lua/cloak/init.lua:19]:
```suggestion
  cloak_telescope = true,
```

**Review by laytan — COMMENTED** [review]
Thanks! Left a few minor reviews

**Comment by eslam-allam** [comment]
Thanks alot mate, glad to be of service!

**Pull request merged**